rbac/management/role/relation_api_dual_write_handler.py

```diff
@@ -120,36 +125,76 @@ def replicate_deleted_system_role(self):
         self._replicate(
             ReplicationEventType.DELETE_SYSTEM_ROLE,
             self._create_metadata_from_role(),
-            self._generate_relations_for_role(),
+            self._generate_relations_for_role(list_all_possible_scopes=True),
             [],
         )
 
-    def _generate_relations_for_role(self) -> list[common_pb2.Relationship]:
+    def check_create_admin__platform_relation(self, role, role_scope, list_relations, permissions):
+        """Check system role and create admin and platform system role parent-child relationship."""
+        if role.admin_default:
+            try:
+                parent_uuid = platform_v2_role_uuid_for(
+                    DefaultAccessType.ADMIN,
+                    role_scope,
+                    GlobalPolicyIdService.shared(),
+                )
+                if parent_uuid:
+                    create_parent_child_relationship = role_child_relationship(parent_uuid, self.role.uuid)
+                    list_relations.append(create_parent_child_relationship)
+            except DefaultGroupNotAvailableError:
+                # Default groups may not exist yet during seeding, skip parent relationship
+                logging.warning(f"Default groups may not exist yet during seeding for admin scope {role_scope}")
+                pass
+
+        if role.platform_default:
+            try:
+                parent_uuid = platform_v2_role_uuid_for(
+                    DefaultAccessType.USER,
+                    role_scope,
+                    GlobalPolicyIdService.shared(),
+                )
+                if parent_uuid:
+                    create_parent_child_relationship = role_child_relationship(parent_uuid, self.role.uuid)
+                    list_relations.append(create_parent_child_relationship)
```

Comment:
Re: [this comment](https://github.com/RedHatInsights/insights-rbac/pull/2047#discussion_r2441241137), we definitely need the try/except, but I think the `if` is unnecessary (and the tests pass locally without it).